cheatsheets/Key_Management_Cheat_Sheet.md

```diff
@@ -151,13 +151,15 @@ The generated keys shall be transported (when necessary) using secure channels a
 1. Developers must understand where cryptographic keys are stored within the application. Understand what memory devices the keys are stored on.
 2. Keys must be protected on both volatile and persistent memory, ideally processed within secure cryptographic modules.
 3. Keys should never be stored in plaintext format.
-4. Ensure all keys are stored in cryptographic vault, such as a [hardware security module](https://en.wikipedia.org/wiki/Hardware_security_module) (HSM) or isolated cryptographic service.
+4. Ensure all keys are stored in a cryptographic vault, such as a [hardware security module](https://en.wikipedia.org/wiki/Hardware_security_module) (HSM) or isolated cryptographic service.
 5. If you are planning on storing keys in offline devices/databases, then encrypt the keys using Key Encryption Keys (KEKs) prior to the export of the key material. KEK length (and algorithm) should be equivalent to or greater in strength than the keys being protected.
 6. Ensure that keys have integrity protections applied while in storage (consider dual purpose algorithms that support encryption and Message Code Authentication (MAC)).
```

Comment:
For for FIPS compliance AES and Triple DES, I would suggest recommending "key wrapping" for this purpose, as it was designed for this for this purpose. There generally is "key wrapping" and "key wrapping with padding", the later when you are wrapping keys that is longer than the cipher's block size. (E.g., AES has a cipher block size of 128-bits, so you would generally use "key wrapping with padding" for wrapping AES key sizes of 192 or 256 bits.)

See [NIST SP 800-38F](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-38F.pdf) and https://en.wikipedia.org/wiki/Key_Wrap for details. As the Wikipedia article points out SIV (see RFC 5297) is probably a better replacement, but I doubt SIV is FIPS compliant. Key wrapping also is not relevant for algorithms other than AES or TDEA.

Java's javax.crypto.Cipher class supports this via the 'opmode' of` Cipher.WRAP_MODE` and `Cipher.UNWRAP_MODE`. Some JCE providers (e.g., BouncyCastle) I believe also implemented GCM-SIV (RFC 8452).  Beyond Java, I'm not sure how well either of these are is supported. There are probably some C and C++ implementations, but I'm not sure about things like Python, Perl, .NET C#, etc. YMMV.